### ISSUE TITLE
Clarify training terminology

### ISSUE THREAD

**Issue by aribrill** [ctlearn-project/ctlearn#61]
As noted in #50, the config option Training:num_epochs really refers to the number of intervals between validating on the validation set, not the number of time to run over the entire training set (as the word epochs usually means). Clarify this by changing the name of the config option and/or adding an explanatory comment. Also, the logging output of run_model.py refers to epochs and this should be to be  consistent with any changes.

As also noted in the same issue, in the logging output of run_model.py, for clarity change 

> Number of training steps per validation

to 

> Number of training steps between validations

**Comment by Jsevillamol** [comment]
Solved in ab2bb65537996c384189aa8ea93d5cbac37e4a8c .

I have opted in the end to change `num_epochs` to `num_validations` and modify the comments where appropriate. After that change I did not feel like the change of `num_steps_per_validation` to `num_steps_between_validations` was needed.

For me it looks clear now, but if anybody feels like the terminology is still confusing please leave a comment.
